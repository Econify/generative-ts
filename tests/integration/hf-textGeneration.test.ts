import {
  createHuggingfaceInferenceModelProvider,
  HfTextGenerationTaskApi,
} from "@packages/core";

test("Huggingface - TextGeneration Task", async () => {
  // arrange
  const gpt2 = createHuggingfaceInferenceModelProvider({
    api: HfTextGenerationTaskApi,
    modelId: "gpt2",
  });

  // act
  const response = await gpt2.sendRequest({
    prompt: "Brief History of NY Mets:",
  });

  // assert
  expect(response).toMatchApiSnapshot();
});
