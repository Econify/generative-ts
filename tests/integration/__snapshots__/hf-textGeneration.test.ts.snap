// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Huggingface - TextGeneration Task 1`] = `
[
  {
    "generated_text": "string",
  },
]
`;
